### PR TITLE
[WIP] [new model]: Ovi 1

### PR DESCRIPTION
**Pull request by vgauraha62**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
New model Ovi 1.1 added

## Test Result
All test passed for implementation
---
<details>
<summary> New model Ovi 1.1 implementation updated diffusion models with pipeline, transformer </summary>

- [New model implementation] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ Yes] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [YES ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Review by lishunyang12 — COMMENTED** [review]
## Early WIP Review -- Ovi 1.1 Model Integration

Thanks for the contribution. Since this is marked WIP, here is an early architectural and correctness review. I have focused on issues that are best caught early before the implementation solidifies.

---

### Bug: RoPE dimension split inconsistency (`ovi_1_1_transformer.py`)

In `Ovi11RotaryPosEmbed.__init__` (around line 72), the temporal/height/width dimension split is computed as:
```python
h_dim = w_dim = 2 * (attention_head_dim // 6)
t_dim = attention_head_dim - h_dim - w_dim
```

But in the `forward` method (around line 109), the split used to re-partition the cached freqs is:
```python
split_sizes = [
    self.attention_head_dim - 2 * (self.attention_head_dim // 3),
    self.attention_head_dim // 3,
    self.attention_head_dim // 3,
]
```

These two formulas are **not equivalent** for most values of `attention_head_dim`. For example, with `attention_head_dim=64`:
- `__init__`: h=w=20, t=24 (total=64)
- `forward`: h=w=21, t=22 (total=64)

This will cause `freqs_cos.split(split_sizes, dim=1)` to split along boundaries that do not match how the buffer was concatenated, leading to silently wrong RoPE embeddings. Please unify the dimension calculation into a single helper or store the split sizes as a buffer/attribute computed once in `__init__`.

---

### Bug: `self.dtype` attribute vs `@property` conflict (`ovi_1_1_transformer.py`, lines ~89-93)

The `__init__` assigns `self.dtype = torch.float32` as an instance attribute, but then a `@property` named `dtype` is defined that returns `next(self.parameters()).dtype`. This is problematic:
1. The instance attribute assignment in `__init__` will shadow `nn.Module.dtype` before any parameters exist.
2. The `@property` on the class will generally win over the instance attribute for attribute lookup, but the `self.dtype = torch.float32` line in `__init__` will actually try to **set** through the property (which has no setter), raising an `AttributeError` at construction time.

Recommendation: remove the `self.dtype = torch.float32` line entirely and rely solely on the `@property`.

---

### Bug: `return_dict` logic is inverted (`ovi_1_1_transformer.py`, lines ~111-113)

```python
if return_dict:
    return (output,)
return output
```

The standard convention in diffusers/HF models is: `return_dict=True` returns a dataclass/dict, `return_dict=False` returns a raw tuple. Here `return_dict=True` returns a plain tuple `(output,)`, and `return_dict=False` returns the bare tensor. This is the opposite of the convention. The pipeline calls `self.transformer(**kwargs)[0]` which always indexes into the result, so functionally it works, but the semantics are misleading and will confuse anyone using the model standalone. Consider either following the convention or removing `return_dict` entirely if it is not needed.

---

### Code quality: Hardcoded `512` magic number (`ovi_1_1_transformer.py`, line ~413)

```python
image_context_length = encoder_hidden_states.shape[1] - 512
```

This is the same pattern as the existing WAN transformer, but it is still a fragile hardcoded constant. At minimum, please define it as a class constant (e.g., `TEXT_CONTEXT_LENGTH = 512`) and reference it by name. Ideally, pass it as a constructor argument so I2V variants with different text encoders are supported.

---

### Code quality: `create_transformer_from_config` is unnecessarily verbose (`pipeline_ovi_1_1.py`, lines ~95-131)

The function has 15 sequential `if "key" in config` blocks that could be replaced with:
```python
KNOWN_KEYS = {"patch_size", "num_attention_heads", ...}
kwargs = {k: (tuple(v) if k == "patch_size" else v) for k, v in config.items() if k in KNOWN_KEYS}
return Ovi11Transformer3DModel(**kwargs)
```

This is more maintainable and less error-prone when new config keys are added.

---

### Design: `nn.Dropout` layers in attention modules

`Ovi11SelfAttention` and `Ovi11CrossAttention` both instantiate `nn.Dropout(dropout)`. Since vLLM-Omni is an inference engine, these dropout layers are always in eval mode and are dead code. Other models in this repo (e.g., WAN) do not include dropout. Consider removing them to keep the code consistent and avoid confusion.

---

### Design: I2V mode is scaffolded but incomplete

The pipeline has I2V-related code paths (image loading in `pre_process_func`, `encoder_hidden_states_image`, `added_kv_proj_dim`, etc.) but the `forward` method has `# TODO: I2V mode` and only implements T2V. This is fine for WIP, but please add a clear guard or `raise NotImplementedError("I2V mode not yet supported")` when an image is provided, rather than silently ignoring it.

---

### Minor: Unused `attention_kwargs` parameter

`attention_kwargs` is passed through to `self.transformer()` but the transformer's `forward` method accepts it only to ignore it -- it is never referenced in the body. Either use it or remove it from the signature to avoid confusion.

---

### Minor: Logging inconsistency

The transformer file uses `from vllm.logger import init_logger` while the pipeline file uses `import logging; logging.getLogger(__name__)`. Please pick one approach for consistency within the module -- `init_logger` is the project convention.

---

Overall the architecture looks reasonable and follows the established WAN-family patterns well. The RoPE split mismatch and the `self.dtype` property conflict are the two issues that will cause runtime failures and should be addressed first. Looking forward to seeing this progress.
